GNUmakefile

```diff
@@ -386,6 +386,18 @@ module-versions:
 envoy-library:
 	@$(SHELL) $(CURDIR)/build-support/scripts/envoy-library-references.sh
 
+.PHONY: envoy-regen
+envoy-regen:
+	$(info regenerating envoy golden files)
+	@for d in endpoints listeners routes clusters rbac; do \
+		if [[ -d "agent/xds/testdata/$${d}" ]]; then \
+			find "agent/xds/testdata/$${d}" -name '*.golden' -delete ; \
+		fi \
+	done
+	@go test -tags '$(GOTAGS)' ./agent/xds -update
+	@find "command/connect/envoy/testdata" -name '*.golden' -delete
```

Comment:
Often we don't need to delete golden files, but in this case the target envoy version is in the filename, so we need to delete them so we prune old files for no longer supported versions, is that right?

Reply:
If you remove a test case this will remove the golden file. That was actually my first justification for having it just reset the output directory. It is also necessary on version bumps.